src/WebJobs.Script.WebHost/Controllers/HostController.cs

```diff
@@ -150,6 +154,21 @@ public IActionResult Restart()
             return Ok(_webHostSettings);
         }
 
+        [HttpPost]
+        [Route("admin/host/offline")]
```

Comment:
Open to ideas on the shape of this API. I don't really want an `admin/host/state` resource API because general state manipulation isn't really supported. This is a targeted API for toggling "offline" on/off.

Reply:
This is state manipulation, though, we just restrict the sates you can set. I would still prefer a proper API under the route `admin/host/status/state`. We should probably consider using a PUT instead of POST here (as this is an idempotent operation).  It would be nice to  avoid growing the number of RPC style APIs, that's why I favor a exposing as a resource that can be set, otherwise, next time we need something that impacts the host state, we'll need to expose a new one.

Reply:
I agree with Fabio. Its fine if there are several transitions that are invalid and the server should return bad request for those transitions. I don't see any reason why this should be represented as RPC.

Reply:
Hey Mathew, I feel like you have not really expressed why exposing the state as a resource is a problem? Also I am not a huge fan of the proposal you shared in the above comment because I think it leaks implementation detail unnecessarily.